classes/reportbuilder/local/entities/error.php

```diff
@@ -0,0 +1,389 @@
+            'status',
+            new lang_string('status', 'search_elastic'),
+            $this->get_entity_name()
+        ))
+            ->add_joins($this->get_joins())
+            ->set_type(column::TYPE_TEXT)
+            ->add_field("{$alias}.status")
+            ->add_callback(static function($value, stdClass $row) use ($statusoptions): string {
+                $statustext = $statusoptions[$row->status] ?? $row->status;
+
+                $statusclasses = [
+                    error_model::STATUS_RETRYING => 'badge-info',
+                    error_model::STATUS_FAILED => 'badge-danger',
+                    error_model::STATUS_OBSOLETE => 'badge-secondary',
+                ];
+
+                $class = 'badge ' . ($statusclasses[$row->status] ?? 'badge-secondary');
+                return html_writer::tag('span', $statustext, ['class' => $class]);
+            })
+            ->set_is_sortable(true);
+
+        $columns[] = (new column(
+            'retrycount',
+            new lang_string('retrycount', 'search_elastic'),
+            $this->get_entity_name()
+        ))
+            ->add_joins($this->get_joins())
+            ->set_type(column::TYPE_INTEGER)
+            ->add_field("{$alias}.retrycount")
+            ->add_callback(static function($value, stdClass $row): string {
+                return $value;
+            })
+            ->set_is_sortable(true);
+
+        $columns[] = (new column(
+            'timemodified',
+            new lang_string('timemodified', 'search_elastic'),
+            $this->get_entity_name()
+        ))
+            ->add_joins($this->get_joins())
+            ->set_type(column::TYPE_TIMESTAMP)
+            ->add_field("{$alias}.timemodified")
+            ->add_callback([format::class, 'userdate'])
+            ->set_is_sortable(true);
+
+        $columns[] = (new column(
+            'contentmodified',
+            new lang_string('contentmodified', 'search_elastic'),
+            $this->get_entity_name()
+        ))
+            ->add_joins($this->get_joins())
+            ->set_type(column::TYPE_TIMESTAMP)
+            ->add_field("{$alias}.contentmodified")
+            ->add_callback(static function($value, stdClass $row): string {
+                return $value ? userdate($value) : '-';
+            })
+            ->set_is_sortable(true);
+
+        $columns[] = (new column(
+            'parentid',
+            new lang_string('parentid', 'search_elastic'),
+            $this->get_entity_name()
+        ))
+            ->add_joins($this->get_joins())
+            ->set_type(column::TYPE_TEXT)
+            ->add_field("{$alias}.parentid")
+            ->add_callback(static function($value, stdClass $row): string {
+                return $value ? html_writer::tag('code', s($row->parentid)) : '-';
+            })
+            ->set_is_sortable(true);
+
+        $baseurl = new moodle_url('/search/engine/elastic/errors.php');
+        $columns[] = (new column(
```

Comment:
As far as I understood report builder design. This column shouldn't be in this entity instance. It should go to your report class instead. 

See an example here https://github.com/catalyst/moodle-tool_dynamic_cohorts/blob/MOODLE_404_STABLE/classes/reportbuilder/local/systemreports/rules.php#L148